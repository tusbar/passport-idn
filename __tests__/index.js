const {readFileSync} = require('fs')

const Strategy = require('..')

const example = readFileSync('__tests__/__fixtures__/example.json', 'utf8')

describe('passport-idn', () => {
  it('should export Strategy constructor directly from package', () => {
    expect(Strategy).toBeInstanceOf(Function)
  })

  it('should load a profile completely', () => {
    const strategy = new Strategy({
      clientID: 'ABC123',
      clientSecret: 'secret'
    }, (() => {
    }))

    strategy._oauth2.get = function (url, accessToken, done) {
      if (url !== 'https://anywhere.idn.laposte.fr/oauth/v2/me') {
        done(new Error('wrong url argument'))
        return
      }

      if (accessToken !== 'token') {
        done(new Error('wrong token argument'))
        return
      }

      done(null, example)
    }

    strategy.userProfile('token', (error, profile) => {
      expect(profile).toMatchSnapshot()
    })
  })
})
