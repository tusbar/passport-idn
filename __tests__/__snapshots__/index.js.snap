// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passport-idn should load a profile completely 1`] = `
Object {
  "_json": Object {
    "birthdate": "01-04-1984",
    "contact/city/home": "Paris",
    "contact/country/home": "France",
    "contact/email": "mbl@somewhere.net",
    "contact/phone/cell": "0607080910",
    "contact/postalAddress/home": ";;4 rue Franconville;;Paris;75050",
    "id": "mbl",
    "namePerson": "M LE-JAMBON Paul",
    "namePerson/first": "Paul",
    "namePerson/friendly": "mbl",
    "namePerson/last": "LE-JAMBON",
    "namePerson/prefix": "M",
  },
  "_raw": "{
  \\"id\\": \\"mbl\\",
  \\"namePerson/friendly\\": \\"mbl\\",
  \\"namePerson\\": \\"M LE-JAMBON Paul\\",
  \\"namePerson/prefix\\": \\"M\\",
  \\"namePerson/last\\": \\"LE-JAMBON\\",
  \\"namePerson/first\\": \\"Paul\\",
  \\"birthdate\\": \\"01-04-1984\\",
  \\"contact/postalAddress/home\\": \\";;4 rue Franconville;;Paris;75050\\",
  \\"contact/city/home\\": \\"Paris\\",
  \\"contact/country/home\\": \\"France\\",
  \\"contact/email\\": \\"mbl@somewhere.net\\",
  \\"contact/phone/cell\\": \\"0607080910\\"
}
",
  "emails": Array [
    Object {
      "value": "mbl@somewhere.net",
    },
  ],
  "id": "mbl",
  "name": "M LE-JAMBON Paul",
  "provider": "idn",
  "username": "mbl",
}
`;
